### ISSUE TITLE
Custom ACS 7.2 ContentModel Properties of type "d:content" are not indexed with ASS 2.0.3

### ISSUE THREAD

**Issue by alexanderjakab** [Alfresco/SearchServices#403]
Since SearchServices 2.x **custom** ACS Content-Model Properties of type "d:content" are no longer indexed. This is a different behaviour from previous versions (e.g. ACS 6.2 & ASS 1.4.3) where all ContentModel Properties of type "d:content" of a Repository Node are Indexed by SearchServices. In 2.x only a node´s "cm:content" Property is indexed.

@aborroy my colleague @aitseitz already discussed this issue with you. for our project this is a major blocker for the upgrade to ACS 7.x. i would be grateful for any advice on a workaround or fix for this problem.

i uploaded a sample sdk 4.4 project to reproduce 
[sdk-sample-project.zip](https://github.com/Alfresco/SearchServices/files/9242272/sdk-sample-project.zip)

steps to reproduce:
1) startup the sdk docker setup from the sample project (extract and run ./run.sh build_start)
2) attach a debugger and set a break-point in org/alfresco/repo/web/scripts/solr/NodeContentGet.class:77 to check the indexing of the nodes content
![image](https://user-images.githubusercontent.com/34883948/182569916-639688ca-15f8-40f3-b5f8-46beffe21b89.png)

2) call the sample Webscript from your browser http://localhost:8080/alfresco/s/sample/helloworld
3) in your nodebrowser the following node is created
![node](https://user-images.githubusercontent.com/34883948/182378305-8dcceaa1-c2ea-44f4-b7ea-df3c6c52b217.PNG)
4) the debugger stops when the solr Metadata Tracker requests the Content
![debugger-solr](https://user-images.githubusercontent.com/34883948/182378564-7a52ba9b-b5b8-47f5-9970-f3d261488ee5.png)

observed behaviour:
1) Only the Content for the "cm:content" property is requested by the MetadataTracker
2) The "acme:metadataIndex" property of the node is never indexed.

expected behaviour:
1) the Content for the "cm:content" property is requested by the MetadataTracker
2) the Content for the "acme:metadataIndex" property is requested by the MetadataTracker. This would be the behavoiur we see in previous versions of SearchServices (e.g. 1.4.3)

additional information:
From checking the versions of the Class org.alfresco.solr.SolrInformationServer the behaviour is represented by the following implementation:

SearchServices 2.0.3
https://github.com/Alfresco/SearchServices/blob/master/search-services/alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java#L2714

![image](https://user-images.githubusercontent.com/34883948/182379930-623936d7-4745-4c8f-a160-265e5fb20294.png)

previous SearchServices Version
https://github.com/Alfresco/SearchServices/blob/636815638468a96b3d75fccfdca8a11045e3654e/search-services/alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java#L2443

![image](https://user-images.githubusercontent.com/34883948/182379901-7eda5bf7-0ce5-44ac-830c-a461f3684678.png)

**Comment by aitseitz** [comment]
Additional info from my side:

It looks like the changes in
https://github.com/Alfresco/SearchServices/blob/master/search-services/alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java#L2714
![image](https://user-images.githubusercontent.com/43071828/182394158-0ddfb394-3b67-4862-b80a-92451b7b580f.png)
have been made back in january 2020 and might be accidently be done with merges related to the
consequence of the ContentStore removal.
See:
https://hub.alfresco.com/t5/alfresco-content-services-blog/search-services-2-0-0-release/ba-p/301308
https://alfresco.atlassian.net/jira/jira/browse/SEARCH-1702 (not accessable for us)

**Comment by aitseitz** [comment]
Official JIRA MNT-Ticket:
https://alfresco.atlassian.net/browse/MNT-23154

**Comment by alexanderjakab** [comment]
following this issue only the documentation was added : https://docs.alfresco.com/search-services/latest/upgrade/breaking-changes-1.x/

**Comment by hi-ko** [comment]
@kkono has a very interesting interpretation of the problem, classifying the no longer indexing of arbitrary d:content properties as enhancement.
The newly introduced restriction of indexing only to cm:content for performance reasons I think is overreaching. It could also be called an undocumented feature-removal, which is a problem without work around for many customers.

**Comment by alexanderjakab** [comment]
thx @hi-ko. that is our interpretation as well . the suggested "workaround" in the documentation not only results in reimplementing customizations but also migrating all existing nodes using that model. basically we expect a fix for this ...

**Comment by aborroy** [comment]
That documentation update was only the first step, to avoid potential users upgrading to Search Services 2.x experimenting this issue. We're working on getting this feature back to 2.x version, but it will take a while.
Thanks for your collaboration on this!

**Comment by hi-ko** [comment]
Thanks @aborroy for the clarification. 

We will have to wait or consider a downgrade to ASS 1.x. The work around described in the doc will not help us in many cases if the additional content properties should be handeled as an integral part of the node. It would be a bottomless pit to solve this with associations to hidden nodes and behaviors taking care of the consistency ...

**Comment by alexanderjakab** [comment]
thx @aborroy for your update

**Comment by aitseitz** [comment]
@alexanderjakab 
Refering to the Release Notes of Alfresco Search Services 2.0.5

![image](https://user-images.githubusercontent.com/43071828/202205784-4374f57c-33ba-4ed3-89da-c9db3974a13d.png)

The https://alfresco.atlassian.net/browse/MNT-23154
was fixed with this ASS 2.0.5.1 release:

https://community.hyland.com/products/alfresco/release-notes/release-notes/alfresco-search-services-module-releases/alfresco-search-services-version-205

If I'm correct the exact code merge commit is

https://github.com/Alfresco/SearchServices/commit/b23578788ce7cf862ebe56b8818198c0b5b91619#diff-486bbf8e33aa92f77d955556566f862a0178db577259e8391d323e98791b5a74R240

which gives us the important information, that we need to change the property
```
# set it to true to allow custom d:content fields to be indexed.
# N.B. At most one content field can be indexed for each document
solr.enableIndexingCustomContent=true
```
in the 
templates/rerank/conf/solrcore.properties
if we need custom "d:content" fields in the solr index!


See:
https://github.com/Alfresco/SearchServices/blob/b23578788ce7cf862ebe56b8818198c0b5b91619/search-services/alfresco-search/src/main/resources/solr/instance/templates/rerank/conf/solrcore.properties#L240
